### PR TITLE
Add block proposal logic for slashing protection

### PR DESCRIPTION
**Pull request by dapplion**
Adds block logic for a refactored slashing protection module compatible with the "Slashing Protection Database Interchange Format"

This PR is part of https://github.com/ChainSafe/lodestar/compare/dapplion/slashing-protection/root, see that branch for a reference full implementation

### CODE REVIEW

File: packages/lodestar-validator/src/slashingProtection/utils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {Epoch, Root} from "@chainsafe/lodestar-types";
+import {toHexString} from "@chainsafe/ssz";
+import {ZERO_HASH} from "@chainsafe/lodestar-beacon-state-transition";
+
+export function isEqualRoot(root1: Root, root2: Root): boolean {
+  return toHexString(root1) === toHexString(root2);
```
Comment (mpetrunic) [inline packages/lodestar-validator/src/slashingProtection/utils.ts:6]: Think it's better/faster to use config.types.Root.equals as it doesn't convert to string first and then compare strings.

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/lodestar-validator/src/slashingProtection/utils.ts (hunk 1 above)
Comment (dapplion) [reply packages/lodestar-validator/src/slashingProtection/utils.ts:6]: Good point! Could you link the implementation of `equals` for Root type?

Comment (mpetrunic) [reply packages/lodestar-validator/src/slashingProtection/utils.ts:6]: not 100% but https://github.com/ChainSafe/ssz/blob/master/src/backings/byteArray/abstract.ts#L20

Comment (dapplion) [reply packages/lodestar-validator/src/slashingProtection/utils.ts:6]: Suggestion applied in https://github.com/ChainSafe/lodestar/pull/1674/commits/1c10190d60378a812ea4a3be81873e054c46493c

**Comment by dapplion** [comment]
Superseded by https://github.com/ChainSafe/lodestar/pull/1674

**Pull request closed without merging**